File: backend/app/schemas/watchlist.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class WatchlistMarketSummary(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    title: str
    slug: str
    category: str
    status: str
    last_price: float
    probability_change: float


class WatchlistItemRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    market_id: int
    position: int | None = None
    notes: str | None = None
    created_at: datetime
    market: WatchlistMarketSummary


class WatchlistRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    created_at: datetime
    updated_at: datetime
    items: list[WatchlistItemRead] = Field(default_factory=list)


class WatchlistCreate(BaseModel):
    name: str = Field(min_length=1, max_length=120)


class WatchlistItemCreate(BaseModel):
    market_id: int
    position: int | None = None
    notes: str | None = None

File: frontend/src/types/watchlist.ts
export type WatchlistMarketSummary = {
  id: number;
  title: string;
  slug: string;
  category: string;
  status: string;
  last_price: number;
  probability_change: number;
};

export type WatchlistItem = {
  id: number;
  market_id: number;
  position: number | null;
  notes: string | null;
  created_at: string;
  market: WatchlistMarketSummary;
};

export type Watchlist = {
  id: number;
  name: string;
  created_at: string;
  updated_at: string;
  items: WatchlistItem[];
};

export type SavedView = {
  id: number;
  name: string;
  layout_json: Record<string, unknown> | null;
  filters_json: Record<string, unknown> | null;
  created_at: string;
  updated_at: string;
};
